lib/uri/common.rb

```diff
@@ -829,7 +829,13 @@ def self.get_encoding(label)
 module Kernel
 
   #
-  # Returns +uri+ converted to an URI object.
+  # Returns a new \URI object derived from the given +uri+,
```

Comment:
It does not necessarily a new URI object. If `uri` is a URI object, then it returns `uri` itself and doesn't create a new one.

Reply:
Fixed.